backend/webserver/views.py

```diff
@@ -67,3 +67,46 @@ class LogoutView(APIView):
     def post(self, request):
         logout(request)
         return Response({'message': 'Successfully Logged out'}, status=status.HTTP_200_OK)
+
+
+
+
+class PostView(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_posts(self,post_id,author_id):
+        post = get_object_or_404(Post,id=post_id,author=author_id)
+        return post
+    
+    def get(self,request,pk,post_id, *args, **kwargs):
+        author =self.get_author(pk=pk)
+        posts = self.get_posts(post_id,author.id)
```

Comment:
Specification says - `GET [local, remote] get the public post whose id is POST_ID`. Shouldn't we be returning only a public post here?